tests/test_hppa.c

```diff
@@ -0,0 +1,179 @@
+/* Capstone Disassembler Engine */
+/* By Dmitry Sibirtsev <sibirtsevdl@gmail.com>, 2023 */
+
+#include <stdio.h>
+
+#include <capstone/platform.h>
+#include <capstone/capstone.h>
+
+struct platform {
+	cs_arch arch;
+	cs_mode mode;
+	unsigned char *code;
+	size_t size;
+	char *comment;
+};
+
+static csh handle;
+
+static void print_string_hex(char *comment, unsigned char *str, size_t len)
+{
+	unsigned char *c;
+
+	printf("%s", comment);
+	for (c = str; c < str + len; c++) {
+		printf("0x%02x ", *c & 0xff);
+	}
+
+	printf("\n");
+}
+
+static void print_insn_detail(cs_insn *ins)
+{
+	cs_hppa *hppa;
+	int i;
+
+	// detail can be NULL on "data" instruction if SKIPDATA option is turned ON
+	if (ins->detail == NULL)
+		return;
+
+	hppa = &(ins->detail->hppa);
+	if (hppa->op_count)
+		printf("\top_count: %u\n", hppa->op_count);
+
+	for (i = 0; i < hppa->op_count; i++) {
+		cs_hppa_op *op = &(hppa->operands[i]);
+		switch ((int)op->type) {
+		default:
+			break;
+        case HPPA_OP_REG:
+            printf("\t\toperands[%u].type: REG = %s\n", i, cs_reg_name(handle, op->reg));
+            break;
+        case HPPA_OP_IMM:
+            printf("\t\toperands[%u].type: IMM = 0x%" PRIx64 "\n", i, op->imm);
+            break;
+        case HPPA_OP_IDX_REG:
+            printf("\t\toperands[%u].type: IDX_REG = %s\n", i, cs_reg_name(handle, op->reg));
+            break;
+        case HPPA_OP_DISP:
+            printf("\t\toperands[%u].type: DISP = 0x%" PRIx64 "\n", i, op->imm);
+            break;
+        case HPPA_OP_MEM:
+            printf("\t\toperands[%u].type: MEM\n", i);
+            if (op->mem.space != HPPA_OP_INVALID) {
+                printf("\t\t\toperands[%u].mem.space: REG = %s\n",
+						i, cs_reg_name(handle, op->mem.space));
+            }
+            printf("\t\t\toperands[%u].mem.base: REG = %s\n",
+                    i, cs_reg_name(handle, op->mem.base));
+            break;
+        case HPPA_OP_TARGET:
+            printf("\t\toperands[%u].type: ", i);
+            if (op->imm >= 0x8000000000000000)
```

Comment:
For the maximum and minimum values of types you can use the macros defined in `platform.h`